File: 0008 String to Integer/StringtoInteger.cs
public class Solution {
    public int MyAtoi(string s) {
        long ret = 0;
        bool isReading = false;
        bool isPostive = true;

        for (int i = 0; i < s.Length; i++)
        {
            char c = s[i];
            if (c == ' ')
            {
                if (isReading) break;
            }
            else if (c == '+')
            {
                if (isReading) break;
                else isReading = true;
            }
            else if (c == '-')
            {
                if (isReading) break;
                else
                {
                    isReading = true;
                    isPostive = false;
                }
            }
            else if ('0' <= c && c <= '9')
            {
                isReading = true;

                ret *= 10;
                if (isPostive) ret += c - '0';
                else ret -= c - '0';

                if (ret < int.MinValue) return int.MinValue;
                else if (int.MaxValue < ret) return int.MaxValue;
            }
            else break;
        }
        return (int)ret;
    }
}

File: 0008 String to Integer/StringtoInteger.py
class Solution:
    def myAtoi(self, s: str) -> int:
        ret = 0
        isReading = False
        isPostive = True
        
        for _, c in enumerate(s):
            if c == ' ':
                if isReading:
                    return ret
            elif c == '+':
                if isReading:
                    return ret
                else:
                    isReading = True
            elif c == '-':
                if isReading:
                    return ret
                else:
                    isReading = True
                    isPostive = False
            elif '0' <= c and c <= '9':
                isReading = True

                ret *= 10
                if isPostive:
                    ret += ord(c) - 48
                else:
                    ret -= ord(c) - 48

                if ret < -2147483648:
                    return -2147483648
                elif 2147483647 < ret:
                    return 2147483647
            else:
                return ret
        return ret